lib/agent/settings.js

```diff
@@ -261,4 +249,167 @@ parseSettings._.checkCookieSettings = function (connectionSettings) {
   return check;
 };
 
+parseSettings.AUTH = {
+  CERT: {
+    PEM_OBJECT_KEYS: {
+      ORDER: 1,
+      SOURCE: "SETTINGS",
+      MANDATORY_KEYS: ["auth.cert", "auth.key"],
+      OPTIONAL_KEYS: ["auth.ca"],
+      ADDITIONAL_KEYS: {
+        "auth.type": "cert",
+      },
+    },
+    PFX_OBJECT_KEYS: {
+      ORDER: 2,
+      SOURCE: "SETTINGS",
+      MANDATORY_KEYS: ["auth.pfx", "auth.passphrase"],
+      OPTIONAL_KEYS: ["auth.ca"],
+      ADDITIONAL_KEYS: {
+        "auth.type": "cert",
+      },
+    },
+    PEM_ENVIRONMENT_KEYS: {
+      ORDER: 3,
+      SOURCE: "ENV",
+      MANDATORY_KEYS: ["ODATA_CLIENT_CERT", "ODATA_CLIENT_KEY"],
+      OPTIONAL_KEYS: ["ODATA_EXTRA_CA"],
+      CONVERSION: {
+        ODATA_CLIENT_CERT: "auth.cert",
+        ODATA_CLIENT_KEY: "auth.key",
+        ODATA_EXTRA_CA: "auth.ca",
+      },
+      ADDITIONAL_KEYS: {
+        "auth.type": "cert",
+      },
+    },
+    CA_OBJECT_KEYS: {
+      ORDER: 4,
+      SOURCE: "SETTINGS",
+      MANDATORY_KEYS: ["auth.ca"],
+      OPTIONAL_KEYS: [],
+    },
+    CA_ENVIRONMENT_KEYS: {
+      ORDER: 5,
+      SOURCE: "ENV",
+      MANDATORY_KEYS: ["ODATA_EXTRA_CA"],
+      OPTIONAL_KEYS: [],
+      CONVERSION: { ODATA_EXTRA_CA: "auth.ca" },
+    },
+  },
+};
+
+parseSettings._.determineTLSDefinition = function (
+  templateDefinitions,
+  connectionSettings,
+  processEnv
+) {
+  return _.map(templateDefinitions, (def, key) =>
+    _.assign(
+      {
+        key: key,
+        source: def.SOURCE === "ENV" ? processEnv : connectionSettings,
+      },
+      def
+    )
+  )
+    .sort((defA, defB) => defA.ORDER - defB.ORDER)
+    .find((def) => def.MANDATORY_KEYS.some((key) => _.has(def.source, key)));
+};
+
+parseSettings._.checkTLSDefinition = function (
```

Comment:
If I'am not missing something, this function is not used, it is redefined on ln 349 and the second definition is used.

Reply:
You are right. It was my copy&paste issue.